.github/workflows/build-and-deploy.yaml

```
@@ -9,11 +9,61 @@ on:
       - main  
   workflow_dispatch:
 
+env:
+  GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+
+
 jobs:
+  reduce-images:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+    
+    - name: Check for large image files
+      id: large-img-check
+      run: |
+        img_path=$(du -ah --threshold=1M | awk '/.(JPE?G|jpe?g|PNG|png|HEIC|heic)$/ {print $2}')
+        echo "$img_path"
+        if [[ -n "$img_path" ]]; then
+          echo "big_image=true" >> $GITHUB_OUTPUT
+          echo "img_path=$(echo $img_path)" >> $GITHUB_OUTPUT
+        fi
+    - name: Install ImageMagick
+      if: steps.large-img-check.outputs.big_image == 'true'
+      run: |
+        cd ..
+        wget https://imagemagick.org/archive/binaries/magick -o magick
+        chmod +x magick
+
+    - name: Shrink images
+      if: steps.large-img-check.outputs.big_image == 'true'
+      env:
+        img_path: ${{ steps.large-img-check.outputs.img_path }}
+      run: |
+        for image in $img_path; do
+          mogrify -resize 800x800 -quality 75 "$image"
```

```suggestion
          mogrify -resize 800x -quality 75 "$image"
```
Only restricting image width to 800 pixels

---

Changed in c2ae1ddd591b21cb61479a01cca8914ed7fb301b